File: services/palace-mcp/tests/extractors/fixtures/py-mini-project/src/pymini/cache.py
from __future__ import annotations

from typing import Generic, Optional, TypeVar

K = TypeVar("K")
V = TypeVar("V")


class Cache(Generic[K, V]):
    def __init__(self) -> None:
        self._store: dict[K, V] = {}

    def put(self, key: K, value: V) -> None:
        self._store[key] = value

    def get(self, key: K) -> Optional[V]:
        return self._store.get(key)

File: services/palace-mcp/tests/extractors/fixtures/ts-mini-project/src/Cache.ts
export class Cache<K, V> {
  private store: Map<K, V>;

  constructor() {
    this.store = new Map<K, V>();
  }

  put(key: K, value: V): void {
    this.store.set(key, value);
  }

  get(key: K): V | undefined {
    return this.store.get(key);
  }
}
